lib/mongoose_push/application.ex

```diff
@@ -4,19 +4,126 @@ defmodule MongoosePush.Application do
   @moduledoc false
```

Comment:
For me, the whole pools management doesn't belong to this module. If would prefer just having it separated out to sth like MongoosePush.Pools which would provide some API to get workers names to start then under supervisors and then the other one to select a woker. Even better, I believe, is a solution in which you have a separate supervisors for each type of pool and it knows more details and the the "pools" module that know even more details.